llvm-external-projects/iree-dialects/include/iree-dialects/Dialect/VectorExt/IR/VectorExtAttrs.td

```diff
@@ -138,164 +138,182 @@ def NestedLayoutAttr : IREEVectorExt_Attr<"NestedLayout",
+      subgroup_layout_order = [0, 1]
 
-    The total number of threads used (computed by multiplying each dim in
-    threads_per_outer) should be a multiple of subgroup size of the
-    harware. If the total number of threads used exceeds the subgroup size of
-    the hardware, then the threads used (say tid) is tid mod subgroup_size:
+      thread_layout_shape = [2, 32]
+      thread_layout_order = [0, 1]
 
-    subgroup_size = 4
+      dim_order = [1, 0]
+    }
 
-    0 1                0 0
-    2 3    tid mod 4   1 1
-    4 5    -------->   2 2
-    6 7                3 3
+    mfma_layout_C = {
+      subgroups_per_workgroup = [2, 2]
+      threads_per_subgroup = [16, 32]
+      elements_per_thread = [4, 1]
 
-    3. Elements per Thread
+      subgroup_layout_shape = [2, 2]
+      subgroup_layout_order = [1, 0]
 
-    The final level of tiling, representing the minimum shape of vector that
-    is treated as an atom.
+      thread_layout_shape = [2, 32]
+      thread_layout_order = [0, 1]
 
-    The elements are placed contigiously with their shape and ordering
-    determined by:
-      - `elements_per_thread`: Sizes of this level of tiling
-      - `element_order`: Ordering of dimensions, from outermost to innermost
+      dim_order = [1, 0]
+    }
   }];
-
+      
   let parameters = (ins
     ArrayRefParameter<"int64_t", "subgroups_per_workgroup">:$subgroupsPerWorkgroup,
-    ArrayRefParameter<"int64_t", "subgroup_order">:$subgroupOrder,
-
-    ArrayRefParameter<"int64_t", "batches_per_subgroup">:$batchesPerSubgroup,
-    ArrayRefParameter<"int64_t", "batch_order">:$batchOrder,
+    ArrayRefParameter<"int64_t", "threads_per_subgroup">:$threadsPerOuter,
+    ArrayRefParameter<"int64_t", "elements_per_thread">:$elementsPerThread,
 
-    ArrayRefParameter<"int64_t", "outers_per_batch">:$outersPerBatch,
-    ArrayRefParameter<"int64_t", "outer_order">:$outerOrder,
+    ArrayRefParameter<"int64_t", "subgroup_layout_shape">:$subgroupLayoutShape,
+    ArrayRefParameter<"int64_t", "subgroup_layout_order">:$subgroupLayoutOrder,
 
-    ArrayRefParameter<"int64_t", "threads_per_outer">:$threadsPerOuter,
-    ArrayRefParameter<"int64_t", "thread_order">:$threadOrder,
+    ArrayRefParameter<"int64_t", "thread_layout_shape">:$threadLayoutShape,
+    ArrayRefParameter<"int64_t", "thread_layout_order">:$threadLayoutOrder,
 
-    ArrayRefParameter<"int64_t", "elements_per_thread">:$elementsPerThread,
-    ArrayRefParameter<"int64_t", "element_order">:$elementOrder
+    ArrayRefParameter<"int64_t", "dim_order">:$dimOrder,
   );
 
-  // TODO: add custom parser/printer and builder to elide default value array
-  // refs.
   let assemblyFormat = [{
     `<` `subgroups_per_workgroup` `=` `[` $subgroupsPerWorkgroup `]` `,`
-        `batches_per_subgroup`    `=` `[` $batchesPerSubgroup `]` `,`
-        `outers_per_batch`        `=` `[` $outersPerBatch `]` `,`
-        `threads_per_outer`       `=` `[` $threadsPerOuter `]` `,`
```

Comment:
We can't drop the `batches_per_subgroup` and `outers_per_batch` here, e.g. https://github.com/openxla/iree/blob/0895f113a977408dab3ff88c9da4839ed2d39df4/compiler/src/iree/compiler/Codegen/Common/GPU/test/gpu_nested_layout_contract_amdgpu.mlir#L407-L420

Reply:
I don't see why?

Reply:
outer_order != batch_order and outer_order != element_order => we need all three (+ their shapes)

Reply:
The example is written a bit wrong. outer_order here doesn't matter because the shape is 4, 1. So the iteration order is same.

Reply:
You don't need the batch and outer shapes. This is just mfma's interpretation of things. You simply need a thread to be holding multiple things. Then you can split them up into batch/outer.